app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show destroy]
+
+  def index
+    @reservations = Reservation.where(user_id: @current_user_id).includes([game: %i[teams stadium]])
+    render json: @reservations, each_serializer: ReservationsSerializer, status: :ok
+  end
+
+  def show
+    render json: @reservation, serializer: ReservationsSerializer
+  end
+
+  def create
+    @game = Game.find(params[:reservation][:game_id])
+    @valid = @game.available_seats >= params[:reservation][:reserved_seats].to_i
+    if @valid
+      @reservation = Reservation.new(reservation_params)
+      @reservation = @game.reservations.create(reservation_params)
+      @reservation.user = @current_user_id
+      if @reservation.save
+        render json: @reservation, status: :created
+      else
+        render json: @reservation.errors.full_messages, status: :unprocessable_entity
+      end
+    else
+      render json: { error: 'Not enough seats available' }, status: :unprocessable_entity
+    end
```

Comment:
Consider DRY and SOLID rules, what cna be done to make this code cleaner
?

Reply:
done